inc/wpseo-non-ajax-functions.php

```diff
@@ -12,298 +12,86 @@
 }
 
 /**
- * Adds an SEO admin bar menu with several options. If the current user is an admin he can also go straight to several settings menu's from here.
+ * Initializes the admin bar.
+ *
+ * @return void
  */
-function wpseo_admin_bar_menu() {
-	// If the current user can't write posts, this is all of no use, so let's not output an admin menu.
-	if ( ! current_user_can( 'edit_posts' ) ) {
-		return;
-	}
-
-	$options = WPSEO_Options::get_options( array( 'wpseo', 'wpseo_ms' ) );
+function wpseo_initialize_admin_bar() {
+	$admin_bar_menu = new WPSEO_Admin_Bar_Menu();
+	$admin_bar_menu->register_hooks();
+}
+add_action( 'wp_loaded', 'wpseo_initialize_admin_bar' );
 
-	if ( $options['enable_admin_bar_menu'] !== true ) {
-		return;
+/**
+ * Allows editing of the meta fields through weblog editors like Marsedit.
+ *
+ * @param array $allcaps Capabilities that must all be true to allow action.
+ * @param array $cap     Array of capabilities to be checked, unused here.
+ * @param array $args    List of arguments for the specific cap to be checked.
+ *
+ * @return array $allcaps
+ */
+function allow_custom_field_edits( $allcaps, $cap, $args ) {
+	// $args[0] holds the capability.
```

Comment:
I was looking for a way to document this in phpdoc. Found this:
```php
         * @param array $args    List of arguments for the specific cap to be checked.
         *  $args = [
	 *      '0'     => (string) holds the capability.
	 *      '1'      => (string) holds the post ID.
	 *      '2'     => (string) holds the custom field.
	 *    ]
```

Maybe that's something to consider. 
For reference: https://stackoverflow.com/questions/15414103/best-way-to-document-array-options-in-phpdoc

Reply:
I don't think it makes sense to document the function parameters like that, since they can be anything, as this function is run for _every_ capability check. However I made the code in the function easier to read and documented it in a way so that it is easier to understand.